### PR TITLE
build(webpack): parcel to webpack

### LINKED ISSUE #14380: Switch from Parcel to Webpack
Parcel does not have WASM support.  Webpack does.  This is a blocker for any work that requires the Rust flux parser.

### PR DESCRIPTION
**Pull request by 121watts**
Closes #14380 

### Pros
- We got our HMR back!  There's even more we can do (like preserving state in between hot-reloads!) 
- Similar if not faster dev and prod build times
- The wasm support we need

### Cons
- Configuration is 💩 

### Notes
For local development we will use `localhost:8080` (the route to the dev server).  All other folks that build for production can still use the lovely 9999.

### CODE REVIEW

File: ui/webpack.dev.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+export {}
```
Comment (chnn) [inline ui/webpack.dev.ts]: what's this for?

File: ui/webpack.common.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+const GIT_SHA = require('child_process')
+  .execSync('git rev-parse HEAD')
+  .toString()
+
+const devMode = process.env.NODE_ENV !== 'production'
+
+module.exports = {
+  context: __dirname,
+  output: {
+    filename: '[name].bundle.js',
```
Comment (chnn) [inline ui/webpack.common.ts]: We should probably add a hash in production.

File: ui/src/shared/components/EditableName.tsx [hunk 3]
```diff
@@ -33,7 +33,7 @@ interface State {
 @ErrorHandling
 class EditableName extends Component<Props, State> {
   public static defaultProps = {
-    hrefValue: '#',
+    hrefValue: 'javascript:void(0);',
```
[the remark this answers is not in the document]
Comment (121watts) [reply ui/src/shared/components/EditableName.tsx]: The `#` was causing a route change to `/#` which, for some reason, causes the app to reload and re-render at the root of the project.

[the remark this answers is not in the document]
Comment (ebb-tide) [reply ui/src/shared/components/EditableName.tsx]: shall we change it everywhere then?

File: ui/webpack.common.ts (hunk 2 above)
Comment (121watts) [reply ui/webpack.common.ts]: Good catch added

**Pull request merged**